test/e2e/auth/selfsubjectattributesreviews.go

```diff
@@ -0,0 +1,119 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("SelfSubjectAttributesReview [Alpha][Feature:APISelfSubjectAttributesReview]", func() {
+	f := framework.NewDefaultFramework("selfsubjectattributesreviews")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	/*
+		Release: v1.24
+		Testname: SelfSubjectAttributesReview API
+		Description:
+		The authentication.k8s.io API group MUST exist in the /apis discovery document.
+		The authentication.k8s.io/v1alpha1 API group/version MUST exist in the /apis/mode.k8s.io discovery document.
+		The selfsubjectattributesreviews resource MUST exist in the /apis/authentication.k8s.io/v1alpha1 discovery document.
+		The selfsubjectattributesreviews resource must support create.
+	*/
+	ginkgo.It("should support SelfSubjectAttributesReview API operations", func() {
+		// Setup
+		ssarAPIVersion := "v1alpha1"
+
+		// Discovery
+		ginkgo.By("getting /apis")
+		{
+			discoveryGroups, err := f.ClientSet.Discovery().ServerGroups()
+			framework.ExpectNoError(err)
+			found := false
+			for _, group := range discoveryGroups.Groups {
+				if group.Name == authenticationv1alpha1.GroupName {
+					for _, version := range group.Versions {
+						if version.Version == ssarAPIVersion {
+							found = true
+							break
+						}
+					}
+				}
+			}
+			if !found {
+				framework.Failf("expected SelfSubjectAttributesReview API group/version, got %#v", discoveryGroups.Groups)
+			}
+		}
+
+		ginkgo.By("getting /apis/authentication.k8s.io")
+		{
+			group := &metav1.APIGroup{}
+			err := f.ClientSet.Discovery().RESTClient().Get().AbsPath("/apis/authentication.k8s.io").Do(context.TODO()).Into(group)
+			framework.ExpectNoError(err)
+			found := false
+			for _, version := range group.Versions {
+				if version.Version == ssarAPIVersion {
+					found = true
+					break
+				}
+			}
+			if !found {
+				framework.Failf("expected SelfSubjectAttributesReview API version, got %#v", group.Versions)
+			}
+		}
+
+		ginkgo.By("getting /apis/authentication.k8s.io/" + ssarAPIVersion)
+		{
+			resources, err := f.ClientSet.Discovery().ServerResourcesForGroupVersion(authenticationv1alpha1.SchemeGroupVersion.String())
+			framework.ExpectNoError(err)
+			found := false
+			for _, resource := range resources.APIResources {
+				switch resource.Name {
+				case "selfsubjectattributesreviews":
+					found = true
+				}
+			}
+			if !found {
+				framework.Failf("expected selfsubjectattributesreviews, got %#v", resources.APIResources)
+			}
+		}
+
+		// Check creating
+		ginkgo.By("creating")
+		{
+			// Use impersonate to make user attributes predictable
+			config := f.ClientConfig()
+			config.Impersonate.UserName = "test-user-selfattributesreview"
+			config.Impersonate.Groups = []string{"system:authenticated", "selfattributesreview-test-group"}
```

Comment:
> @deads2k - I can't remember, is the user running e2e tests expected to be able to impersonate arbitrarily?

I think they are expected to be cluster-admins, so that any feature in the API can be tested.  I would expect this to also allow impersonation.